src/services/cart.service.ts

```diff
@@ -12,61 +12,62 @@ export class CartService {
     this.loadOrderlines();
   }
 
-  loadOrderlines(cb = null) {
-    this.httpService.post(`cart/items`, {data: {}}).subscribe(
-      data => {
-        this.updateInfo(data);
-        this.dataArray = data;
-        if (cb) cb();
-      }, err => {
-        console.error("err", err);
-      }
-    );
+  loadOrderlines() {
+    return new Promise((resolve, reject) => {
+      this.httpService.post(`cart/items`, {data: {}}).subscribe(
+        data => {
+          this.updateInfo(data);
+          this.dataArray = data;
+          resolve();
+        },
+        err => {
+          console.error("error in loading orderlines:", err);
+          reject();
+        }
+      );
+    });
   }
 
   updateInfo(data) {
     this.dataArray = data;
     this.cartItems.next(this.getTotalNumber());
   }
 
-  addOrderline(product_id, product_instance_id, number, cb) {
-    let data = {product_id, product_instance_id, number};
-    this.httpService.post(`order`, data).subscribe(
-      res => {
-        // this.dataArray.push({
-        //   instance_id: product_instance_id,
-        //   quantity: number,
-        // });
-        // this.cartItems.next(this.getTotalNumber());
-        this.loadOrderlines();
-        if (!(res.n > 0 || res.nModified > 0))
-          return cb('nothing is changed');
-
-        if (cb) cb(null);
-      }, err => {
-        console.error('error in adding orderline', err);
-        if (cb) cb(err);
-      }
-    );
+  addOrderline(product_id, product_instance_id, number) {
+    return new Promise((resolve, reject) => {
+      let data = {product_id, product_instance_id, number};
+      this.httpService.post(`order`, data).subscribe(
+        res => {
+          this.loadOrderlines();
+          if(res.n <= 0 && res.nModified <= 0)
+            return reject();
```

Comment:
No recommended to use 'return reject()';
Recommended to use return Promise.reject() or reject if no accepted block after it (Because reject not preform as return statement: return stop continuous code to run but reject no)